modules/publication/ajax/FileDelete.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Publication file deletion script
+ *
+ * Deletes a specified file by upload ID (PublicationUploadID)
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Publication
+ * @author   David <dblader.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+$uploadID = $_REQUEST['uploadID'];
+$db       = \Database::singleton();
+$user     = \User::singleton();
+$config   = \NDB_Config::singleton();
+
+$query      = "SELECT PublicationID, URL ".
+    "FROM publication_upload ".
+    "WHERE PublicationUploadID=:upid";
+$uploadData = $db->pselectRow($query, array('upid' => $uploadID));
+
+if (empty($uploadData)) {
+    header("HTTP/1.1 400 Bad Request");
+    return;
+}
+
+userCanDelete($uploadData, $db, $user);
```

Comment:
This function only changes the http header, and it'll still fall through to the delete below if the user isn't supposed to be able to delete it.